packages/@aws-cdk/pipelines/lib/blueprint/script-step.ts

```diff
@@ -0,0 +1,271 @@
+import { CfnOutput, Stack } from '@aws-cdk/core';
+import { mapValues } from '../private/javascript';
+import { FileSet, IFileSetProducer } from './file-set';
+import { StackDeployment } from './stack-deployment';
+import { Step } from './step';
+
+/**
+ * Construction properties for a `ScriptStep`.
+ */
+export interface ScriptStepProps {
+  /**
+   * Commands to run
+   */
+  readonly commands: string[];
+
+  /**
+   * Installation commands to run before the regular commands
+   *
+   * For deployment engines that support it, install commands will be classified
+   * differently in the job history from the regular `commands`.
+   *
+   * @default - No installation commands
+   */
+  readonly installCommands?: string[];
+
+  /**
+   * Environment variables to set
+   *
+   * @default - No environment variables
+   */
+  readonly env?: Record<string, string>;
+
+  /**
+   * Set environment variables based on Stack Outputs
+   *
+   * `ScriptSteps` following stack or stage deployments may
+   * access the `CfnOutput`s of those stacks to get access to
+   * --for example--automatically generated resource names or
+   * endpoint URLs.
+   *
+   * @default - No environment variables created from stack outputs
+   */
+  readonly envFromOutputs?: Record<string, CfnOutput>;
```

Comment:
Maybe. I did not do that because `output.value` is not a thing, and I wasn't convinced it makes sense in any other context than here. It doesn't make a whole lot of sense in CloudFormation, typically (which is why we didn't have it yet).

I mean I see we can supposedly render it to a `Fn::ImportValue` (though that turns the `Output` into an `Export` which are not the same).

All in all, there seems risk in that decision for limited upside, which is why I didn't go for that yet.

But we can always still add it later.